### DISCUSSION TITLE
mirror another server

### DISCUSSION THREAD

**Discussion by ronytesler** [eclipse-milo/milo#1024] [General]
I want to have an application that will be a client of some upcua servers, but also will be an opcua server (using milo) for its own clients.
I want this application to send to its clients the same data it gets from the opcua servers.
For this, I assume the application needs to have the same nodes as the servers have.
I'd like to add to the application only the nodes that it subscribes to.
Is it possible to add nodes at run time, and make sure they're identical to the nodes in the opcua servers?

**Comment by kevinherron** [comment]
Yes, you can add/remove Nodes at runtime. What you're trying to do should be possible. You may run into issues if the other servers implement OPC UA 1.04/1.05 and are using structured values and DataTypeDefinitions rather than the legacy DataType Dictionary, but if you're just trying to mirror "simple" types there shouldn't be an issue.

**Reply by ronytesler** [reply]
run into issues but still possible to implement, or it would be too complicated?

**Reply by kevinherron** [reply]
Milo only implements OPC UA 1.03 right now. It wouldn't be possible to accurately mirror a server that only used 1.04+ datatypes.
